rfcs/0050-merge-bot-for-maintainers.md

```diff
@@ -0,0 +1,83 @@
+---
+feature: merge-bot-for-maintainers
+start-date: 2019-08-17
+author: Frederik Rietdijk
+co-authors: (find a buddy later to help our with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Allow maintainers of packages and modules in Nixpkgs to submit their own changes
+when certain criteria are fullfilled.
+
+# Motivation
+[motivation]: #motivation
+
+Nixpkgs is growing. More people should be able to help out with merging changes.
+Merging changes typically requires write permission, however, that gives the
+possibility for Nixpkgs-wide changes and decreases the security.
+
+By allowing maintainers of packages and modules to submit their own changes the
+load on members of the @nixpkgs-committers that have write permission group is
+reduced.
+
+# Detailed design
+[design]: #detailed-design
+
+Maintainers of packages and modules will be able to submit their own changes
+when the following criteria are met:
+1. The label `11.by: package-maintainer` is set by @grahamcofborg
+2. The packages/modules are not new.
+3. All packages/modules affected need to be maintained by the maintainer.
+
+When these conditions are met the maintainer can write
+
+    @grahamcofborg merge
+
+Note that also another bot could be used than @grahamcofborg.
+
+Criteria 1) is to be met to ensure that it is indeed the maintainer that wants
+to submit the changeset. It is important that the label is adjusted with the
+requirement set in criteria 3), that is, that all affected derivations, so also
+reverse dependencies, are to be maintained by the maintainer. This is to ensure
+that maintainers cannot make changes elsewhere in the tree and be permitted to
+submit them because they have also made a change in an expression they own. This
+also means that this process nearly only applies to leaf packages. Criteria 2)
+exists to ensure that at least the initial expression was reviewed by a member
+of @nixpkgs-maintainers.
```

Comment:
Doesn't it make more sense to say that a member of nixpkgs-commiters reviewed it, since they're responsible for what gets integrated into the codebase?

Reply:
What part exactly are you referring to?

Reply:
> exists to ensure that at least the initial expression was reviewed by a member of nixpkgs-maintainers. 

That should be nixpkgs-committers right?

Reply:
Correct, that is a mistake in the RFC.